### PR TITLE
feat: Scene 3

### PR DESCRIPTION
**Pull request by ivanma9**
Add scene3 background 
Add foregrounds of color

### CODE REVIEW

File: src/components/pages/Story1/index.tsx [hunk 1]
```diff
@@ -3,18 +3,39 @@ import { useHistory } from 'react-router';
...
+      expo: [
+        {
+          text: 'george why so lonely go play ball ',
+          duration: 3000,
+        },
+        {
+          text: 'so slow no wonder u in online school',
+          duration: 2000,
+        },
+      ],
```
Comment (BryanPan342) [inline src/components/pages/Story1/index.tsx]: pls add the text required for this scene

File: src/components/shared/Layout.tsx [hunk 2]
```diff
@@ -15,6 +15,11 @@ export interface SceneProps {
   expo: Exposition[];
   image: string;
   imageAlt?: string;
+  style?: {
+    width?: string,
+    margin?: string,
+    border?: string,
+  };
```
Comment (BryanPan342) [inline src/components/shared/Layout.tsx]: i think we can just do smth like `CSS.Properties`

File: src/components/sketches/Scene3.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import {
+  INIT_PARTICLES_COUNT,
+  INIT_SPEED,
+  ACCEL,
+} from '../../utils/constants';
+
+export default function sketch(p: p5): void {
+  let canvas: p5.Renderer;
+  const particles: Particle[] = [];
+  const max = 150;
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]: turn this into a constant?

File: src/components/sketches/Scene3.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  let BALL_RADIUS:number;
+
+  let BOUNDARY_RADIUS:number;
+  let BOUNDARY_X:number;
+  let BOUNDARY_Y:number;
+  const PINK: p5.Color = p.color(158, 34, 94);
+  const HOT_PINK:p5.Color = p.color(184, 47, 170);
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]:
```suggestion
  let BALL_RADIUS: number;

  let BOUNDARY_RADIUS: number;
  let BOUNDARY_X: number;
  let BOUNDARY_Y: number;
  const PINK: p5.Color = p.color(158, 34, 94);
  const HOT_PINK: p5.Color = p.color(184, 47, 170);
```

File: src/components/sketches/Scene3.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    canvas.id('p5-background');
+    BALL_RADIUS = p.width / 100;
+    BOUNDARY_RADIUS = p.width * 0.32;
+    BOUNDARY_X= p.width * 0.4;
+    BOUNDARY_Y = p.height * 0.7;
+    // Setting mode to degrees for spawning particles in circle
+    p.angleMode(p.DEGREES);
+    p.background(0);
+    // Initialize array
+    // Max number of particles
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]: remove these comments bc they are unneccessary?

File: src/components/sketches/Scene3.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  const PINK: p5.Color = p.color(158, 34, 94);
+  const HOT_PINK:p5.Color = p.color(184, 47, 170);
+
+  p.setup = () => {
+    canvas = p.createCanvas(window.innerWidth, window.innerHeight);
+    canvas.id('p5-background');
+    BALL_RADIUS = p.width / 100;
+    BOUNDARY_RADIUS = p.width * 0.32;
+    BOUNDARY_X= p.width * 0.4;
+    BOUNDARY_Y = p.height * 0.7;
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]: Why can't we just make BALL_RADIUS const and calculate using `window.innerWidth`?

File: src/components/sketches/Scene3.ts [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+    // Setting mode to degrees for spawning particles in circle
+    p.angleMode(p.DEGREES);
+    p.background(0);
+    // Initialize array
+    // Max number of particles
+    p.frameRate(30);
+
+    for (let i = 0; i < INIT_PARTICLES_COUNT; i++) {
+      particles.push(new Particle(p.random(p.width), p.random(p.height), p.random(0, 359), BALL_RADIUS, PINK));
+    }
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]: nit: convert to map

File: src/components/sketches/Scene3.ts [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  };
+  p.windowResized = () => {
+    p.resizeCanvas(window.innerWidth, window.innerHeight, true);
+  };
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]:
```suggestion
  };

  p.windowResized = () => {
    p.resizeCanvas(window.innerWidth, window.innerHeight, true);
  };
```

File: src/components/sketches/Scene3.ts [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+      // Delete old particles (from beginning of array)
+      particles.splice(0, 10);
+    }
+
+    // Loop through the array and show each particle
+    for (let i = 0; i < particles.length; i++) {
+      particles[i].update();
+      particles[i].show();
+
+    }
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]:
```suggestion
    }
```

File: src/components/sketches/Scene3.ts [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+  p.mouseClicked = () => {
+    particles.push(new Particle(p.mouseX, p.mouseY, p.random(0, 359), BALL_RADIUS, HOT_PINK));
+  };
+
+  class Particle {
+    public pos: p5.Vector;
+    private _vel: p5.Vector;
+    private _acc: p5.Vector;
+    private _radius: number;
+    public color:p5.Color;
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]:
```suggestion
    public pos: p5.Vector;
    public color: p5.Color;
    private _vel: p5.Vector;
    private _acc: p5.Vector;
    private _radius: number;
```

File: src/components/sketches/Scene3.ts [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+        this._acc.x *= -1;
+        this._vel.x *= -1;
+        this.pos.x = 0 + this._radius;
+        this._vel.add(this._acc);
+      } else if (this.pos.x > p.width - this._radius) {
+        this._acc.x *= -1;
+        this._vel.x *= -1;
+        this.pos.x = p.width - this._radius;
+        this._vel.add(this._acc);
+      }
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]:
```suggestion
      if (this.pos.x < this._radius || this.pos.x > p.width - this._radius) {
        this._acc.x *= -1;
        this._vel.x *= -1;
        this.pos.x = this.pos.x > p.width - this._radius ? p.width - this._radius : this._radius;
        this._vel.add(this._acc);
      }
```

File: src/components/sketches/Scene3.ts [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+      if (this.pos.y < 0 + this._radius) {
+        this._acc.y *= -1;
+        this._vel.y *= -1;
+        this.pos.y = 0 + this._radius;
+        this._vel.add(this._acc);
+      } else if (this.pos.y > p.height - this._radius) {
+        this._acc.y *= -1;
+        this._vel.y *= -1;
+        this.pos.y = p.height - this._radius;
+        this._vel.add(this._acc);
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]: same change as the one above

File: src/components/sketches/Scene3.ts [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+      } else if (p.sqrt(x_dist * x_dist + y_dist * y_dist) < BOUNDARY_RADIUS / 2) {
+        this._vel.reflect(p.createVector(x_dist, y_dist));
+      } else {
+        if (p.sqrt(this._vel.x * this._vel.x + this._vel.y * this._vel.y) > INIT_SPEED) {
+          this._vel.limit(INIT_SPEED + 2);
+        }
+      }
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]: is this supposed to be in another if/else bracket? and if so pls label it to describe its behavior

**Review by BryanPan342 — CHANGES_REQUESTED** [review]
Looking really clean! Got a couple of comments but other than that this is super clean

- Scene4 png shouldnt be in this pr

File: src/components/shared/Layout.tsx (hunk 2 above)
Comment (ivanma9) [reply src/components/shared/Layout.tsx]: Ok I will take a stab

File: src/components/sketches/Scene3.ts (hunk 13 above)
Comment (ivanma9) [reply src/components/sketches/Scene3.ts]: These series, represent whether the ball hits a wall or the circle, so its all one chain to ensure direction change of one wall only. I will add comments tho

File: src/components/pages/Story1/index.tsx (hunk 1 above)
Comment (ivanma9) [reply src/components/pages/Story1/index.tsx]: I think Forrest may set up a pr for this. I'll check, if not I will put it in for all the scenes (1,3,4)

File: src/components/shared/Layout.tsx [hunk 14]
```diff
@@ -15,6 +16,7 @@ export interface SceneProps {
   expo: Exposition[];
   image: string;
   imageAlt?: string;
+  style?: CSS.Properties;
```
Comment (BryanPan342) [inline src/components/shared/Layout.tsx]: Does this work?

```suggestion
  style?: React.CSSProperties;
```

File: src/components/shared/Layout.tsx [hunk 15]
```diff
@@ -91,7 +93,7 @@ function Layout(props: LayoutProps): JSX.Element {
   return (
     <div id={'layout'}>
       <P5Scene sketch={scene.sketch}/>
-      <img src={scene.image} alt={scene.imageAlt} className={'foreground-image'}/>
+      <img src={scene.image} alt={scene.imageAlt} className={'foreground-image'} style ={scene.style}/>
```
Comment (BryanPan342) [inline src/components/shared/Layout.tsx]:
```suggestion
      <img src={scene.image} alt={scene.imageAlt} className={'foreground-image'} style={scene.style}/>
```

File: package.json [hunk 16]
```diff
@@ -10,6 +10,7 @@
   },
   "dependencies": {
     "animejs": "^3.2.1",
+    "csstype": "^3.0.8",
```
Comment (BryanPan342) [inline package.json]: remove

File: src/components/sketches/Scene3.ts [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+
+    public edges() {
+      const x_dist = this.pos.x - BOUNDARY_X;
+      const y_dist = this.pos.y - BOUNDARY_Y;
+      // If hitting edge, invert appropriate velocity
+      if (this.pos.x < this._radius || this.pos.x > p.width - this._radius) {
+        this._acc.x *= -1;
+        this._vel.x *= -1;
+        this.pos.x =
+					this.pos.x > p.width - this._radius? p.width - this._radius : this._radius;
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]:
```suggestion
        this.pos.x = this.pos.x < this._radius ? this._radius : p.width - this._radius;
```

File: src/components/sketches/Scene3.ts [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+        this._vel.add(this._acc);
+      }
+      else if (this.pos.y < this._radius || this.pos.y > p.height - this._radius) {
+        this._acc.y *= -1;
+        this._vel.y *= -1;
+        this.pos.y =
+					this.pos.y > p.height - this._radius? p.height - this._radius : this._radius;
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]:
```suggestion
        this.pos.y = this.pos.y < this._radius ? this._radius : p.height - this._radius;
```

File: src/components/sketches/Scene3.ts [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+      } else if
+      /** Bounce if touching Boundary */
+      (p.sqrt(x_dist * x_dist + y_dist * y_dist) < BOUNDARY_RADIUS / 2) {
+        this._vel.reflect(p.createVector(x_dist, y_dist));
+      } else {
+        /** If not hitting a wall ensure speed does not go past INIT_SPEED + 2 */
+        if (
+          p.sqrt(this._vel.x * this._vel.x + this._vel.y * this._vel.y) >
+					INIT_SPEED
+        )
```
Comment (BryanPan342) [inline src/components/sketches/Scene3.ts]: this should be an else if

**Review by BryanPan342 — CHANGES_REQUESTED** [review]
Please remove the scene4.png file

File: src/components/sketches/Scene3.ts (hunk 6 above)
Comment (ivanma9) [reply src/components/sketches/Scene3.ts]: we originally did, but I am trying to resize the balls based off window resizing and it wasnt working; it still doesnt resize unless I refresh the page...might need some assistance there.

File: src/components/sketches/Scene3.ts (hunk 7 above)
Comment (ivanma9) [reply src/components/sketches/Scene3.ts]: maps have not been working on these initialization stuff

Comment (ivanma9) [reply src/components/sketches/Scene3.ts]: @forrestburton @BryanPan342 do you guys wanna take a stab at changing to map?

File: src/components/shared/Layout.tsx (hunk 14 above)
Comment (ivanma9) [reply src/components/shared/Layout.tsx]: it says React has no exported member CSS

Comment (ivanma9) [reply src/components/shared/Layout.tsx]: but it still works, so ig its good

File: src/components/shared/Layout.tsx [hunk 20]
```diff
@@ -15,6 +15,7 @@ export interface SceneProps {
   expo: Exposition[];
   image: string;
   imageAlt?: string;
+  style?: React.CSS.Properties;
```
Comment (BryanPan342) [inline src/components/shared/Layout.tsx]:
```suggestion
  style?: React.CSSProperties;
```

Comment (BryanPan342) [reply src/components/shared/Layout.tsx]: this is the right way to do it @ivanma9

File: src/components/sketches/Scene3.ts (hunk 7 above)
Comment (BryanPan342) [reply src/components/sketches/Scene3.ts]: its the same problem as the other one

look at line 11
```
const particles: Particle[] = [];
```

this const is blocking you

File: yarn.lock [hunk 21]
```diff
@@ -3172,7 +3172,7 @@ cssesc@^3.0.0:
   resolved "https://registry.yarnpkg.com/cssesc/-/cssesc-3.0.0.tgz#37741919903b868565e1c09ea747445cd18983ee"
   integrity sha512-/Tb/JcjK111nNScGob5MNtsntNM1aCNUDipB/TkwZFhyDrrE47SOx/18wF2bbjgc3ZzCSKW1T5nt5EbFoAz/Vg==
 
-csstype@^3.0.2:
+csstype@^3.0.2, csstype@^3.0.8:
```
Comment (BryanPan342) [inline yarn.lock]: run yarn install again!

**Review by BryanPan342 — COMMENTED** [review]
One last thing!

**Review by BryanPan342 — APPROVED** [review]
LGTM!!!

**Pull request merged**